src/runtime/runtime_cortexm.go

```diff
@@ -100,21 +104,57 @@ type interruptStack struct {
 //go:export handleHardFault
 func handleHardFault(sp *interruptStack) {
 	print("fatal error: ")
-	if uintptr(unsafe.Pointer(sp)) < 0x20000000 {
-		print("stack overflow")
+
+	if machine.CPUArchitecture == "armv7m" || machine.CPUArchitecture == "armv7em" || machine.CPUArchitecture == "armv8m" {
```

Comment:
I first tried `if a := machine.CPUArchitecture; a == "armv7m" || a == "armv7em" || a == "armv8m" { ... }`, but the compiler did not optimize away the branch.